components/scream/src/physics/share/physics_test_data.hpp

```diff
@@ -0,0 +1,157 @@
+#ifndef SCREAM_PHYSICS_TEST_DATA_HPP
+#define SCREAM_PHYSICS_TEST_DATA_HPP
+
+#include "share/scream_types.hpp"
+
+#include "ekat/util/ekat_math_utils.hpp"
+#include "ekat/ekat_assert.hpp"
+
+#include <vector>
+#include <utility>
+
+//
+// PhysicsTestData is meant to offer the client something they can inherit to provide
+// convenient handling of arrays of data in the common *Data structs that are used for
+// unit-testing and bridging. This class supports up to 4 classes of data
+//   * dim1 of Real
+//   * (dim1 x dim2) of Real
+//   * (dim1 x dim3) of Real
+//   * dim1 of Int
+//
+// Subclasses of PhysicsTestData should look like the following. Note that the copy
+// constructor and assignment operators must be defined if you want to be able to copy
+// objects of this type. The DATA_COPY_CONS and ASSIGNX macros are there to help you do this.
+/*
+struct SHOCGridData : public PhysicsTestData {
+  // Inputs
+  Real *zt_grid, *zi_grid, *pdel;
+
+  // In/out
+  Real *dz_zt, *dz_zi, *rho_zt;
+
+  SHOCGridData(Int shcol_, Int nlev_, Int nlevi_) :
+    PhysicsTestData(shcol_, nlev_, nlevi_,
+      {&zt_grid, &dz_zt, &pdel, &rho_zt},  // list of (shcol x nlev) members
+      {&zi_grid, &dz_zi}) {}               // list of (shcol x nlevi) members
+
+  DATA_COPY_CONS(SHOCGridData, 3); // 3 => number of arguments constructor expects
+  ASSIGN0(SHOCGridData); // 0 => number of scalars
+};
+*/
+
+// Convenience macros for up to 10 arguments, beyond that, you're on your own :)
+
+#define REP0()
```

Comment:
Do you expect the `ASS*` and `REP*` macros to be used outside this file, or only by the preprocessor in expanding the `ASSIGN` and `DATA_COPY_CONS` macros below? If you don't need them for anything else, maybe we should #undef them after we define those macros.

Reply:
That is a good thought. I tried that myself and discovered it doesn't work because those macros need to be available when the ASSIGN macro is used.

Reply:
Curses! In that case, maybe Luca's comment about prepending `SCREAM_` or maybe `PTEST_` to these macros to make them more specific (as I think Luca might have suggested) might make this less of an issue. After all, their actual names only appear in this file.

Reply:
Interesting. I was assuming that macros were expanded at file parsing in the order they appear. So I thought that the REP* macros would be immediately expanded in the _definition_ of  `DATA_COPY_CONS`, and the following expansions of `DATA_COPY_CONS` would not include `REP*` anymore.

Reply:
Oh oh oh, I see why. The CPP does not know which REP* will be called when it finds the definition of `DATA_COPY_CONS`, so it can't replace `REP*` yet. Makes sense.

Reply:
Yeah, I "namespaced" all the macros with "PTD_" (for PhysicsTestData).